File: xianguoji-server/src/main/java/com/xianguoji/server/module/order/entity/Order.java
package com.xianguoji.server.module.order.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Data
@TableName("`order`")
public class Order {

    @TableId(type = IdType.AUTO)
    private Long id;
    private String orderNo;
    private Long userId;
    private Integer status;
    private Integer payStatus;
    private Integer deliveryType;
    private String deliveryTime;
    private Long addressId;
    private String consignee;
    private String consigneePhone;
    private String consigneeAddress;
    private Long pickupPointId;
    private String pickupCode;
    private BigDecimal goodsAmount;
    private BigDecimal couponAmount;
    private BigDecimal discountAmount;
    private BigDecimal deliveryFee;
    private BigDecimal payAmount;
    private Long userCouponId;
    private String payMethod;
    private LocalDateTime payTime;
    private String payTradeNo;
    private String userRemark;
    private String cancelReason;
    private Long groupBuyInstanceId;
    private String courierName;
    private String courierPhone;
    private LocalDateTime deliveredAt;
    private LocalDateTime finishedAt;
    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createdAt;
    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updatedAt;
    @TableField(fill = FieldFill.INSERT)
    private Long createBy;
    @TableField(fill = FieldFill.INSERT_UPDATE)
    private Long updateBy;
}

File: merchant-front/src/api/types/order.ts
/**
 * 后端 OrderVO（com.xianguoji.server.module.order.vo.OrderVO）
 * 商家端订单复用同一个 VO（AdminOrderController 返回 OrderVO）
 */
export interface AdminOrderVO {
  id: number;
  orderNo: string;
  status: number;
  payStatus: number;
  deliveryType: 1 | 2;
  deliveryTime?: string;
  consignee: string;
  consigneePhone: string;
  consigneeAddress: string;
  pickupCode?: string;
  goodsAmount: string;
  couponAmount: string;
  discountAmount: string;
  deliveryFee: string;
  payAmount: string;
  userRemark?: string;
  cancelReason?: string;
  groupBuyInstanceId?: number;
  payTime?: string;
  deliveredAt?: string;
  finishedAt?: string;
  createdAt: string;
  items: AdminOrderItemVO[];
}

export interface AdminOrderItemVO {
  id: number;
  productId: number;
  skuId: number;
  productName: string;
  specName: string;
  image: string;
  price: string;
  originalPrice: string;
  quantity: number;
  subtotal: string;
  isReviewed: number;
}

export interface AdminOrderQry {
  status?: number;
  keyword?: string;
  startDate?: string;
  endDate?: string;
  page?: number;
  size?: number;
}

export interface ShipDto {
  /** 1=配送 2=自提 */
  deliveryType?: number;
  courierName?: string;
  courierPhone?: string;
}
